test/extended/operators/cluster.go

```diff
@@ -0,0 +1,251 @@
+package operators
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/beorn7/perks/quantile"
+
+	"github.com/beorn7/perks/histogram"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = g.Describe("[Feature:Platform][Suite:openshift/smoke-4] Managed cluster should", func() {
+	defer g.GinkgoRecover()
+
+	g.It("have no crashlooping pods in core namespaces over two minutes", func() {
+		c, err := e2e.LoadClientset()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		var lastPodsWithProblems []*corev1.Pod
+		var pending map[string]*corev1.Pod
+		wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+			allPods, err := c.Core().Pods("").List(metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
if this includes bootstrapping I wouldn't count on not seeing an error or two during pivot.

Reply:
I suspect we’ll have to tune this.  I was mostly focused on how we can add more comprehensive e2e tests that verify invariants rather than test function - things we generally *want* to be true, but sometimes aren’t.

The real challenge will be when this starts flaking.  Do we take the time to improve our api latency?  Do we look for ways to smooth out blocking changes in the master?  Or do we just bump the limits and shrug?

It’s certainly hard to define strict rules.  We got massively better in perf from 3.0 - 3.7, then gradually slid as we split and added hops.  Was the split beneficial in other ways?  Sure.  Did we measure the improvement and hold the split to the same standards of not regressing customer experience?  In general it’s good for us to ask these meta questions and get some cautious limits in place, even if we decide they are too aggressive later.